go-controller/pkg/config/config.go

```diff
@@ -1,105 +1,575 @@
+	CNI.Plugin = getConfigStr(cliConfig.CNI.Plugin, cfg.CNI.Plugin, "ovn-k8s-cni-overlay")
+
+	Kubernetes.Kubeconfig = getConfigStr(cliConfig.Kubernetes.Kubeconfig, cfg.Kubernetes.Kubeconfig, "")
+	if Kubernetes.Kubeconfig != "" && !pathExists(Kubernetes.Kubeconfig) {
+		return fmt.Errorf("kubernetes kubeconfig file %q not round", Kubernetes.Kubeconfig)
+	}
+	Kubernetes.CACert = getConfigStr(cliConfig.Kubernetes.CACert, cfg.Kubernetes.CACert, "")
+	if Kubernetes.CACert != "" && !pathExists(Kubernetes.CACert) {
+		return fmt.Errorf("kubernetes CA certificate file %q not round", Kubernetes.CACert)
+	}
+	Kubernetes.Token = getConfigStr(cliConfig.Kubernetes.Token, cfg.Kubernetes.Token, "")
+	Kubernetes.APIServer = getConfigStr(cliConfig.Kubernetes.APIServer, cfg.Kubernetes.APIServer, "https://localhost:8443")
+	if _, err = url.Parse(Kubernetes.APIServer); err != nil {
+		return fmt.Errorf("kubernetes API server address %q invalid: %v", Kubernetes.APIServer, err)
+	}
+
+	OvnNorth.ClientAuth, OvnNorth.ServerAuth, err = getOvnAuth(true, &cliConfig.OvnNorth, &cfg.OvnNorth)
+	if err != nil {
+		return err
+	}
+
+	OvnSouth.ClientAuth, OvnSouth.ServerAuth, err = getOvnAuth(false, &cliConfig.OvnSouth, &cfg.OvnSouth)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("Default config: %+v", Default)
+	logrus.Debugf("Logging config: %+v", Logging)
+	logrus.Debugf("CNI config: %+v", CNI)
+	logrus.Debugf("Kubernetes config: %+v", Kubernetes)
+	logrus.Debugf("OVN North config: %+v", OvnNorth)
+	logrus.Debugf("OVN South config: %+v", OvnSouth)
+
+	return nil
+}
+
+// OvnDBAuth describes an OVN database location and authentication method
+type OvnDBAuth struct {
+	URL     string
+	PrivKey string
+	Cert    string
+	CACert  string
+
+	server bool
+	host   string
+	port   string
+	Scheme OvnDBScheme
+}
+
+func pathExists(path string) bool {
+	_, err := os.Stat(path)
+	if err != nil && os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+// newOvnDBAuth returns an OvnDBAuth object describing the connection to an
+// OVN database, given a connection description string and authentication
+// details
+func newOvnDBAuth(ctlCmd, urlString, privkey, cert, cacert string, server bool) (*OvnDBAuth, error) {
+	if urlString == "" {
+		if privkey != "" || cert != "" || cacert != "" {
+			return nil, fmt.Errorf("certificate or key given; perhaps you mean to use the 'ssl' scheme?")
+		}
+		return &OvnDBAuth{
+			server: server,
+			Scheme: OvnDBSchemeUnix,
+		}, nil
+	}
+
+	url, err := url.Parse(urlString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse OVN DB URL %q: %v", urlString, err)
+	}
+	host, port, err := net.SplitHostPort(url.Host)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse OVN DB host/port %q: %v", url.Host, err)
+	}
+	// OVN requires the --db argument to be an IP, not a DNS name
+	ip := net.ParseIP(host)
+	if ip == nil {
+		return nil, fmt.Errorf("OVN DB host %q must be an IP address, not a DNS name", url.Host)
```

Comment:
Would it be possible to just resolve the name to an IP for the user?

Reply:
@pecameron it might, but we'd have to do that internally.  This isn't a regression though, this statement was present before this PR.